File: app/src/main/java/com/example/rssreader/data/RssDatabase.java
package com.example.rssreader.data;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.support.annotation.NonNull;

import com.example.rssreader.App;

@Database(entities = {RssSource.class}, version = 1)
public abstract class RssDatabase extends RoomDatabase {
    private static RssDatabase instance;

    public abstract RssSourceDao sourceDao();

    @NonNull
    public static RssDatabase instance() {
        if (instance == null) {
            instance = Room
                    .databaseBuilder(App.getAppContext(), RssDatabase.class, "db")
                    .allowMainThreadQueries() // todo remove me
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/example/rssreader/data/db/RssDatabase.kt
package com.example.rssreader.data.db

import androidx.room.Database
import androidx.room.RoomDatabase
import com.example.rssreader.domain.entity.RssSource

@Database(entities = [RssSource::class], version = 1)
abstract class RssDatabase : RoomDatabase() {
    abstract fun sourceDao(): RssSourceDao
}